File: C++/Estrutura condicional/Baskara.cpp
#include <iostream>
#include <iomanip>
#include <string>
#include <climits>
#include <cmath>

using namespace std;

int main()
{

    double a, b, c, x1, x2, delta;

    cout << "Coeficiente a: ";
    cin >> a;
    cout << "Coeficiente b: ";
    cin >> b;
    cout << "Coeficiente c: ";
    cin >> c;

    delta = pow(b, 2) - 4 * a * c;

    if (a == 0 || delta < 0)
    {
        cout << "Esta equacao nao possui raizes reais" << endl;
    }
    else
    {
        x1 = (-b + sqrt(delta)) / (2 * a);
        x2 = (-b - sqrt(delta)) / (2 * a);
        cout << fixed << setprecision (4);
        cout << "X1 =" << x1 << endl;
        cout << "X2 =" << x2 << endl;
    }

    return 0;
}

File: Python/Estrutura Condicional/Baskara.py
import math

a: float; b: float; c: float; x1 = float; x2 = float; delta: float

a = float(input("Coeficiente a: "))
b = float(input("Coeficiente b: "))
c = float(input("Coeficiente c: "))


delta = math.pow(b, 2.0) - 4 * a * c

if (a == 0) != (delta < 0):
    print(f"Esta equacao nao possui raizes reais")
else:
    x1 = (-b + math.sqrt(delta)) / (2 * a)
    x2 = (-b - math.sqrt(delta)) / (2 * a)
    print(f"X1 = {x1:.4f}")
    print(f"X2 = {x2:.4f}")

